### PR TITLE
fix Menu Items (closes #1232)

### LINKED ISSUE #1232: Chat options in menu are enabled when no group is selected
when no group is selected, the chat options like 'delete chat', 'block contact', etc are enabled, causing errors if user clicks them, this option should be disabled, or better: hidden/removed from the menu if there are no chat selected

### PR DESCRIPTION
**Pull request by Simon-Laux**
refactor menu and chatListItem contextmenu for ease of development
hide some options that don't make any sense on selftalk and deviceTalk,
in mainMenu and chatListItem contextmenu

### CODE REVIEW

File: src/main/deltachat/chatlist.js [hunk 1]
```diff
@@ -99,6 +99,8 @@ module.exports = class DCChatList extends SplitOut {
       freshMessageCounter: this._dc.getFreshMessageCount(chatId),
       isArchiveLink: chat.id === C.DC_CHAT_ID_ARCHIVED_LINK,
       contactIds,
+      isSelfTalk: chat.isSelfTalk || undefined, // only transfer if needed (json strips undefined)
+      isDeviceTalk: chat.isDeviceTalk || undefined,
```
Comment (Jikstra) [inline src/main/deltachat/chatlist.js]: Why strip it out? I don't think that we need to make the json strings smaller for any performance improvements.

Comment (nicodh) [reply src/main/deltachat/chatlist.js]: I would also prefer to always have it false or true and not undefined....

Comment (Simon-Laux) [reply src/main/deltachat/chatlist.js]: wasn't space saving the whole idea of chatlist item instead of fullchat?

Comment (nicodh) [reply src/main/deltachat/chatlist.js]: No the idea was not to fetch to many unused data (causing more database queries). Memory wise we have no issue with json fields I would assume...

Comment (Simon-Laux) [reply src/main/deltachat/chatlist.js]: then we can do that, but I'm still skeptical how fast the electron ipc can handle much json data fast.

Comment (Jikstra) [reply src/main/deltachat/chatlist.js]: Don't be skeptical, measure it :) As long as there's no measurement, I wouldn't strip stuff out.

**Pull request merged**